src/Xamarin.Android.Build.Tasks/Tasks/ResolveAssemblies.cs

```diff
@@ -56,11 +58,12 @@ bool Execute (DirectoryAssemblyResolver resolver)
 			Log.LogDebugMessage ("  I18nAssemblies: {0}", I18nAssemblies);
 			Log.LogDebugMessage ("  LinkMode: {0}", LinkMode);
 			Log.LogDebugTaskItems ("  Assemblies:", Assemblies);
+			Log.LogDebugTaskItems ("  DesignTimeFacadeDirectories:", DesignTimeFacadeDirectories);
 
 			foreach (var dir in ReferenceAssembliesDirectory.Split (new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries))
 				resolver.SearchDirectories.Add (dir);
 
-			var assemblies = new HashSet<string> ();
+			var assemblies = new HashSet<ITaskItem> ();
```

Comment:
Is this safe or desirable?

The [`TaskItem` documentation](https://msdn.microsoft.com/en-us/library/microsoft.build.utilities.taskitem.aspx) states that `GetHashCode()` and `Equals(Object)` are "(Inherited from Object.)". [The source code for `TaskItem`](https://github.com/Microsoft/msbuild/blob/74f1a4505e07bc4e531e7688de43f3a1ebfa4e00/src/Utilities/TaskItem.cs) likewise doesn't override `GetHashCode()` or `Equals()` or implement `IEquatable<T>`.

Consequently, this should thus be using *referential equality*, which means the following will store *two* values for the same path, not one:

    var assemblies = new HashSet<ITaskItem> {
        new TaskItem ("foo/bar"),
        new TaskItem ("foo/bar"),
    };
    // assemblies.Count == 2

This almost certainly isn't what we want. Perhaps we should use the [`HashSet<T>(IEqualityComparer<T>)`](https://msdn.microsoft.com/en-us/library/bb359100(v=vs.110).aspx) constructor?